### PR TITLE
do not force MonadIO on presign

### PR DESCRIPTION
**Pull request by avieth**

### CODE REVIEW

File: lib/amazonka/src/Amazonka/Presign.hs [hunk 1]
```diff
@@ -11,6 +11,9 @@
 --
```
Comment (endgame) [inline lib/amazonka/src/Amazonka/Presign.hs:11]: Please update the copyright year in the haddock, and remove references to `'MonadIO'` (removed here) and to `AWS specific transformers` (which have been gone for a long time).

File: lib/amazonka/src/Amazonka/Presign.hs [hunk 2]
```diff
@@ -11,6 +11,9 @@
...
 -- It is intended for use directly with "Amazonka.Auth" when only presigning
```
Comment (endgame) [inline lib/amazonka/src/Amazonka/Presign.hs:12]:
```suggestion
-- It is intended for use directly with 'Amazonka.Auth.Auth' when only presigning
```

File: lib/amazonka/src/Amazonka/Presign.hs [hunk 3]
```diff
@@ -11,6 +11,9 @@
...
 -- and no other AWS actions are required.
```
Comment (endgame) [inline lib/amazonka/src/Amazonka/Presign.hs:13]:
```suggestion
-- and no other AWS actions are required. See 'Amazonka.Auth.withAuth' to
-- extract an 'AuthEnv' from an 'Auth'
```

**Review by endgame — CHANGES_REQUESTED** [review]
Since `withAuth` still has to have a `MonadIO` constraint, I'm curious what this is buying you. I see it's coming from a workplace fork, which means it's proven necessary for something, so I'm happy to merge it with a couple of tweaks.

In addition to the suggestions inline, there's a couple more changes I'll need to see before I can merge it, but those are minor:

- [ ] Add an entry to `lib/amazonka/CHANGELOG.md`
- [x] Re-export `withAuth` from `Amazonka.Auth` (as a general rule, most normal users should not have to depend directly on `amazonka-core`)

**Comment by endgame** [comment]
I agree that it's a better factoring, and I hope we can merge it before I tag RC2.

**Comment by avieth** [comment]
Should be good to go now. LMK if I missed anything.

File: lib/amazonka/src/Amazonka/Presign.hs [hunk 4]
```diff
@@ -6,11 +6,12 @@
 -- Stability   : provisional
 -- Portability : non-portable (GHC extensions)
 --
--- This module contains functions for presigning requests using 'MonadIO' and
--- not one of the AWS specific transformers.
---
--- It is intended for use directly with "Amazonka.Auth" when only presigning
--- and no other AWS actions are required.
+-- It is intended for use directly with "Amazonka.Auth.Auth" when only
```
Comment (endgame) [inline lib/amazonka/src/Amazonka/Presign.hs:9]:
```suggestion
-- It is intended for use directly with 'Amazonka.Auth.Auth' when only
```

File: lib/amazonka/src/Amazonka/Presign.hs [hunk 5]
```diff
@@ -6,11 +6,12 @@
 -- Stability   : provisional
```
Comment (endgame) [inline lib/amazonka/src/Amazonka/Presign.hs:6]: Please update copyright year in the haddock.

**Review by endgame — CHANGES_REQUESTED** [review]
Couple of final nits, but I look forward to merging this today if you have time to make the changes.

**Review by endgame — APPROVED** [review]
Thank you very much.

**Pull request merged**